test/quantization/test_quant_api.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+# mypy: ignore-errors
+import unittest
+import torch
+from torch._export import capture_pre_autograd_graph
+from torch.ao.quantization.quantize_pt2e import (
+    prepare_pt2e,
+    convert_pt2e,
+)
+from torch.ao.quantization.quantizer.xnnpack_quantizer import (
+    XNNPACKQuantizer,
+    get_symmetric_quantization_config,
+)
+
+from torchao.quantization.quant_api import _replace_with_custom_fn_if_matches_filter
+from torchao.quantization.quant_api import apply_dynamic_quant
+from torchao.quantization.quant_api import (
+    Quantizer,
+    TwoStepQuantizer,
+)
+
+def dynamic_quant(model, example_inputs):
+    m = capture_pre_autograd_graph(model, example_inputs)
+    quantizer = XNNPACKQuantizer().set_global(get_symmetric_quantization_config(is_dynamic=True))
+    m = prepare_pt2e(m, quantizer)
+    m = convert_pt2e(m)
+    return m
+
+def _apply_dynamic_quant(model):
+    """
+    Applies dynamic symmetric per-token activation and per-channel weight
+    quantization to all linear layers in the given model using
+    module swaps.
+    """
+    _replace_with_custom_fn_if_matches_filter(
+        model,
+        lambda linear_mod: dynamic_quant(linear_mod, (torch.randn(1, linear_mod.in_features))),
+        lambda mod, fqn: isinstance(mod, torch.nn.Linear),
+    )
+    return model
+
+
+def capture_and_prepare(model, example_inputs):
+    m = capture_pre_autograd_graph(model, example_inputs)
+    quantizer = XNNPACKQuantizer().set_global(get_symmetric_quantization_config(is_dynamic=True))
```

Comment:
also torchao is currently oriented for gpu quantization per the blurb, i think it'd be good to be super clear what APIs are for gpu quantization, or more specifically, keep non gpu stuff in a separate folder

Reply:
we need some refactoring for existing quantization apis in https://github.com/pytorch-labs/ao/tree/main/torchao/quantization, I think we should have the separation of quant flow and lowering step for all backends